### PR TITLE
build(jest): upgrade jest + tests

### PR DESCRIPTION
**Pull request by billyvg**
* Breaking changes to tests, `this` no longer works in test suites, declare
variable at outter most `describe` block instead
* `Client` mock has new `mockAsync` static var to mock async responses

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: tests/js/spec/components/assigneeSelector.spec.jsx [hunk 1]
```diff
@@ -28,18 +32,18 @@ describe('AssigneeSelector', function() {
...
+    sandbox.stub(MemberListStore, 'getAll').returns([USER_1, USER_2]);
+    sandbox.stub(GroupStore, 'get').returns({
       id: 1337,
       assignedTo: null
     });
   });
 
   afterEach(function() {
-    this.sandbox.restore();
+    sandbox.restore();
```
Comment (benvinegar) [inline tests/js/spec/components/assigneeSelector.spec.jsx]: Is there not a possibility of the now closure-scoped `sandbox` variable having different values between async test executions?

**Review by benvinegar — COMMENTED** [review]
> Breaking changes to tests, this no longer works in test suites, declare variable at outter most describe block instead

Why'd they make this change? 😤

File: tests/js/spec/components/assigneeSelector.spec.jsx (hunk 1 above)
Comment (billyvg) [reply tests/js/spec/components/assigneeSelector.spec.jsx]: Hmm, don't tests still execute serially regardless if they are async?

**Comment by billyvg** [comment]
ref for `this`: https://github.com/facebook/jest/issues/3628

File: tests/js/spec/components/avatar.spec.jsx [hunk 2]
```diff
@@ -3,18 +3,20 @@ import {shallow} from 'enzyme';
...
 describe('Avatar', function() {
+  let sandbox;
+
   const USER = {
     id: 1,
     name: 'Jane Doe',
     email: 'janedoe@example.com',
   };
 
   beforeEach(function() {
```
Comment (MaxBittker) [inline tests/js/spec/components/avatar.spec.jsx]: it would be nice to have a wrapper or something that reduced a bit of this boilerplate some day

**Review by MaxBittker — APPROVED** [review]
looks good to me, nice upgrade 👍 Not sure i prefer the new `let sandbox` situation but /shrug

**Comment by MaxBittker** [comment]
reading that issue didn't really sell me :P

**Pull request merged**